fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/values/InOpValue.java

```diff
@@ -213,7 +213,6 @@ public InFn() {
         private static Value encapsulateInternal(@Nonnull final List<? extends Typed> arguments) {
             final Typed arg0 = arguments.get(0);
             final Type res0 = arg0.getResultType();
-            SemanticException.check(res0.isPrimitive(), SemanticException.ErrorCode.COMPARAND_TO_COMPARISON_IS_OF_COMPLEX_TYPE);
```

Comment:
The following code never took a record type into account. Can you provide a test case that tests that you cannot do an e.g. 
```
(a, b) in (1, 2, 3)
(a, b) in ((1, 2, 3), (4, 5, 6))
```